spec/scrabble_spec.js

```diff
@@ -1,9 +1,253 @@
+  it('return score of each letter', function() {
+    for(letter in points){
+      expect(scrabble.score(letter)).toEqual(points[letter]);
+    }
+  });
+
+  it('return score of 5 for the word cat', function() {
+      expect(scrabble.score('cat')).toEqual(5);
+
+  });
+
+  it('return score of 57 for the word aaaaaaa', function() {
+      expect(scrabble.score('aaaaaaa')).toEqual(57);
+
+  });
+
+  it('return score of 0 if no word is given', function() {
+      expect(scrabble.score('')).toEqual(0);
+
+  });
+
+});
+
+
+describe('Scrabble#highestScoreFrom(array_of_words)', function() {
+  var scrabble = new Scrabble();
+
+  // build of array of words and return the highest value word
+
+  it("should return the word 'hello' for the word_array = ['hello']", function(){
+    var arrrayOfWords = ["hello"];
+    expect(scrabble.highestScoreFrom(arrrayOfWords)).toEqual("hello");
+  });
+
+  it("should return the word 'hello' as the highest score from the word_array = ['hello', 'cat']", function(){
+    var arrrayOfWords = ["hello", "cat"];
+    expect(scrabble.highestScoreFrom(arrrayOfWords)).toEqual("hello");
+  });
+
+  // # for tiebreaker, the word with less tiles used is returned
+  it("should return the word 'fee' from the word_array = ['cats', 'fee', 'no']", function(){
+    var arrrayOfWords = ["cats", "fee", "no"];
+    expect(scrabble.highestScoreFrom(arrrayOfWords)).toEqual("fee");
+  });
+
+  //  if top score is tied between multiple words but one used all seven letters, that one wins
+  it ("should return the word 'aaaaaaa' from the word_array = ['feet', 'aaaaaad', 'no', 'qqqqqj']", function(){
+    // chose the following words because aaaaaad's score is 58 including the 7-letter bonus,
+    // and qqqqqj's score is also 58 just because of the rare letters
+    var arrayOfWords = ["feet", "aaaaaad", "no", "qqqqqj"];
+    expect(scrabble.highestScoreFrom(arrayOfWords)).toEqual("aaaaaad");
+  });
+
+  it ("should return the word 'aaaaaaa' from the word_array = ['feet', 'qqqqqj, 'no', 'aaaaaad']! BACKWARDS MODE!", function(){
+    // # chose the following words because aaaaaad's score is 58 including the 7-letter bonus,
+    // # and qqqqqj's score is also 58 just because of the rare letters
+    var arrayOfWords = ["feet", "qqqqqj", "no", "aaaaaad"];
+    expect(scrabble.highestScoreFrom(arrayOfWords)).toEqual("qqqqqj");
+  });
+
+  // # testing multiple words with same score and same length
+  it ("should return 'aaaaaaa' from the word_array = ['aaaaaaa', 'eeeeeee', 'frog']", function(){
+    var arrayOfWords = ["aaaaaaa", "eeeeeee", "frog"];
+    expect(scrabble.highestScoreFrom(arrayOfWords)).toEqual("aaaaaaa");
+  });
+
+  it ("should return 'aaaaaad' from the word_array = [''nose'', ''aaaaaad'', ''eeeeeed'', ''frog'', ''rrrrrrr'', ''qqqqqj'']", function(){
+    var arrayOfWords = ["nose", "aaaaaad", "eeeeeed", "frog", "rrrrrrr", "qqqqqj"];
+    expect(scrabble.highestScoreFrom(arrayOfWords)).toEqual("aaaaaad");
+  });
+});
+
```

Comment:
Thanks!!

On Wed, Jun 15, 2016 at 9:55 PM, Jennie Buechner <notifications@github.com>
wrote:

> In spec/scrabble_spec.js
> <https://github.com/Ada-C5/js-scrabble/pull/17#discussion_r67287351>:
>
> > +    var arrayOfWords = ["feet", "qqqqqj", "no", "aaaaaad"];
> > +    expect(scrabble.highestScoreFrom(arrayOfWords)).toEqual("qqqqqj");
> > +  });
> > +
> > +  // # testing multiple words with same score and same length
> > +  it ("should return 'aaaaaaa' from the word_array = ['aaaaaaa', 'eeeeeee', 'frog']", function(){
> > +    var arrayOfWords = ["aaaaaaa", "eeeeeee", "frog"];
> > +    expect(scrabble.highestScoreFrom(arrayOfWords)).toEqual("aaaaaaa");
> > +  });
> > +
> > +  it ("should return 'aaaaaad' from the word_array = [''nose'', ''aaaaaad'', ''eeeeeed'', ''frog'', ''rrrrrrr'', ''qqqqqj'']", function(){
> > +    var arrayOfWords = ["nose", "aaaaaad", "eeeeeed", "frog", "rrrrrrr", "qqqqqj"];
> > +    expect(scrabble.highestScoreFrom(arrayOfWords)).toEqual("aaaaaad");
> > +  });
> > +});
> > +
>
> Thorough testing of many different scenarios 👏
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Ada-C5/js-scrabble/pull/17/files/68f6a6ae8a0ccd97a9fe4f3a205163ad6eefb2ee#r67287351>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AOaZbeoFdQ6B5FMGERuKinj6EWdcP5QRks5qMNdHgaJpZM4Iz-uw>
> .
>